internal/beater/middleware/monitoring_middleware.go

```diff
@@ -18,36 +18,79 @@
 package middleware
 
 import (
+	"context"
 	"net/http"
 
-	"github.com/elastic/elastic-agent-libs/monitoring"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
 
 	"github.com/elastic/apm-server/internal/beater/request"
+	"github.com/elastic/elastic-agent-libs/monitoring"
 )
 
-// MonitoringMiddleware returns a middleware that increases monitoring counters for collecting metrics
-// about request processing. As input parameter it takes a map capable of mapping a request.ResultID to a counter.
-func MonitoringMiddleware(m map[request.ResultID]*monitoring.Int) Middleware {
+type monitoringMiddleware struct {
+	meter metric.Meter
+
+	ints     map[request.ResultID]*monitoring.Int
+	counters map[request.ResultID]metric.Int64Counter
+}
+
+func (m *monitoringMiddleware) Middleware() Middleware {
 	return func(h request.Handler) (request.Handler, error) {
-		inc := func(id request.ResultID) {
-			if counter, ok := m[id]; ok {
-				counter.Inc()
-			}
-		}
 		return func(c *request.Context) {
-			inc(request.IDRequestCount)
+			ctx := context.Background()
+
+			m.getMetric(request.IDRequestCount).Add(ctx, 1)
+			m.inc(request.IDRequestCount)
 
 			h(c)
 
-			inc(request.IDResponseCount)
+			m.getMetric(request.IDResponseCount).Add(ctx, 1)
+			m.inc(request.IDResponseCount)
 			if c.Result.StatusCode >= http.StatusBadRequest {
-				inc(request.IDResponseErrorsCount)
+				m.getMetric(request.IDResponseErrorsCount).Add(ctx, 1)
+				m.inc(request.IDResponseErrorsCount)
 			} else {
-				inc(request.IDResponseValidCount)
+				m.getMetric(request.IDResponseValidCount).Add(ctx, 1)
+				m.inc(request.IDResponseValidCount)
 			}
 
-			inc(c.Result.ID)
+			m.getMetric(c.Result.ID).Add(ctx, 1)
+			m.inc(c.Result.ID)
 		}, nil
 
 	}
 }
+
+func (m *monitoringMiddleware) inc(id request.ResultID) {
+	if counter, ok := m.ints[id]; ok {
+		counter.Inc()
+	}
+}
+
+func (m *monitoringMiddleware) getMetric(n request.ResultID) metric.Int64Counter {
+	name := "http.server." + n
```

Comment:
I named those metrics `http.server.xxx`, and not `apm-server.http.server.xxx`, as the former better matches the otel semantic conventions.
https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md

We can rename those metrics in the local exporter.